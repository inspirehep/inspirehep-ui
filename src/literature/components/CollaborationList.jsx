import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { List } from 'immutable';

import CollaborationLink from './CollaborationLink';
import InlineList from '../../common/components/InlineList';

class CollaborationList extends Component {
  render() {
    const { collaborations, wrapperClassName } = this.props;
    return (
      <InlineList
        wrapperClassName={wrapperClassName}
        items={collaborations}
        extractKey={collaboration => collaboration.get('value')}
        renderItem={collaboration => (
          <CollaborationLink>{collaboration.get('value')}</CollaborationLink>
        )}
      />
    );
  }
}

CollaborationList.propTypes = {
  collaborations: PropTypes.instanceOf(List),
  wrapperClassName: PropTypes.string,
};

CollaborationList.defaultProps = {
  collaborations: null,
  wrapperClassName: null,
};

export default CollaborationList;
