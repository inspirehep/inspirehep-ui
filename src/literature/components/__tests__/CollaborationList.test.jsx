import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import CollaborationList from '../CollaborationList';

describe('CollaborationList', () => {
  it('renders with collaborations', () => {
    const collaborations = fromJS([
      { value: 'Alias Investigations' },
      { value: 'Nelson and Murdock ' },
    ]);
    const wrapper = shallow(
      <CollaborationList collaborations={collaborations} />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
