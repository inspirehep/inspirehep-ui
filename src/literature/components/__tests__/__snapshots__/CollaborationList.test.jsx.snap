// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollaborationList renders with collaborations 1`] = `
<div
  className="__InlineList__"
>
  <ul
    className="separate-items"
  >
    <li
      key="Alias Investigations"
    >
      <CollaborationLink>
        Alias Investigations
      </CollaborationLink>
    </li>
    <li
      key="Nelson and Murdock "
    >
      <CollaborationLink>
        Nelson and Murdock 
      </CollaborationLink>
    </li>
  </ul>
</div>
`;
